cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -647,13 +649,21 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(int podNumber, Strin
         return pvc;
     }
 
-    protected Volume createEmptyDirVolume(String name) {
-        Volume volume = new VolumeBuilder()
-                .withName(name)
-                .withNewEmptyDir()
-                .endEmptyDir()
+    protected Volume createEmptyDirVolume(String name, String sizeLimit) {
+        EmptyDirVolumeSource emptyDirVolumeSource;
+        if (sizeLimit == null || sizeLimit.isEmpty()) {
+            emptyDirVolumeSource = new EmptyDirVolumeSourceBuilder()
+                .build();
+        } else {
+            emptyDirVolumeSource = new EmptyDirVolumeSourceBuilder()
+                .withNewSizeLimit(sizeLimit)
                 .build();
-        log.trace("Created emptyDir Volume named '{}'", name);
+        }
+        Volume volume = new VolumeBuilder()
+            .withName(name)
+            .withEmptyDir(emptyDirVolumeSource)
+            .build();
```

Comment:
What you have is fine, but (as you know because you used it in the tests) you can use the builders to create the entire tree of objects in one expression, like this:

```java
new VolumeBuilder()
  .withName(name)
  .withNewEmptyDir()
    .withNewSizeLimit(sizeLimit.isEmpty() ? null : sizeLimit)
  .endEmptyDir()
.build()
```

That's actually much easier to understand because it reflects the (YAML) structure.

Reply:
Updated as suggested, with a slight tweak to avoid a null pointer if sizeLimit is passed in as null.